core-web/src/main/java/org/silverpeas/core/web/treemenu/process/TreeBuilder.java

```diff
@@ -194,35 +199,45 @@ private static MenuItem buildFirstLevel(TreeFilter filter, String userId, String
     MenuItem item = new MenuItem("root");
     ArrayList<MenuItem> children = new ArrayList<>();
     item.setChildren(children);
-    // the space
     OrganizationController controller = OrganizationControllerProvider.getOrganisationController();
+
+    // the spaces
     if (filter.acceptNodeType(NodeType.SPACE)) {
       List<SpaceInstLight> rootSpaces =
           controller.getRootSpacesContainingComponent(userId, "kmelia");
+      if (TreeHandler.useOrder)
+        rootSpaces.sort(Comparator.comparing(SpaceInstLight::getOrderNum));
+
       for (SpaceInstLight space : rootSpaces) {
         if (space != null) {
           MenuItem subElement =
               new MenuItem(space.getName(language), space.getId(), 0,
-              NodeType.SPACE, false, null, null);
+                  NodeType.SPACE, false, null, null);
           children.add(subElement);
         }
-      }// the component
-    } else if (filter.acceptNodeType(NodeType.COMPONENT) && !filter.getComponents().isEmpty()) {
+      }
+    }
+
+    // the components
+    if (filter.acceptNodeType(NodeType.COMPONENT) && !filter.getComponents().isEmpty()) {
       for (String componentName : filter.getComponents()) {
         List<ComponentInstLight> componentList =
             controller.getAvailComponentInstLights(userId, componentName);
+
+        if (TreeHandler.useOrder)
+          componentList.sort(Comparator.comparing(ComponentInstLight::getOrderNum));
+
```

Comment:
A Java rule: always the body of an if or else-statements is put in brackets, even if there is only a single statement of code

More, this code is duplicated several times: lines 99, 122 and 208. It can be defined as a private method accepting tow parameters: the list to sort and the sorting function (here respectively `NodeDetail::getOrder`, `SpaceInstLight::getOrderNum`, `ComponentInstLight::getOrder`)

Reply:
I need help to factorize a method with differents type parameters: Comparator.comparing(NodeDetail, Comparator.comparing(ComponentInstLight etc..
idem for the list (List<SpaceInstLight>, List<ComponentInstLight>...)